internal/util/GitService.go

```diff
@@ -485,5 +489,23 @@ func (impl GitHubClient) ensureProjectAvailability(projectName string) (validate
 		}
 		time.Sleep(10 * time.Second)
 	}
+	_, err = impl.createReadme(projectName)
```

Comment:
don't call this method from `ensureProjectAvailability`  it breaks single responsibility principle.  
rather create two methods `ensureProjectAvailabilityHttp` and `ensureProjectAvailabilityssh` and call them sequentially.